Add article on Azure DevOps service connection security

BlueBasher opened the pull request:
# Pull Request Template

## What are you trying to address
This PR adds an article that describes a potential security issue in Azure DevOps and how to fix that issue.

## Description of new changes

A new article has been added that describes a security issue that can be present in an Azure DevOps setup regarding Service Connections. The article describes what the actual issue is and how Service Connections should be configured to fix the issue.

## For all pull requests

- [ ] Link to the issue you are solving (so it gets closed)
- [ ] Label the pull request with the appropriate area(s)
- [ ] Assign potential reviewers (you may also want to contact them on Teams to ensure timely reviews)
- [ ] Assign the project - Engineering Playbook Backlog
- [ ] Assign the pull request to yourself

TessFerrandez commented:
Thank you for the PR - generally I think the content is good, but I think perhaps it fits better in Continuous Delivery or Continuous Integration under dev-sec-ops -- there are some similar patterns in there

BlueBasher commented:
Before creating the PR, I was in doubt whether it should be Security or Continuous Integration. But CI seems like a good fit indeed.
I've moved the content and updated the PR.

shiranr commented on docs/continuous-integration/dev-sec-ops/azure-devops/service-connection-security.md:
```diff
@@ -0,0 +1,49 @@
+
+In Azure DevOps, Service Connections are used in Azure DevOps Pipelines to connect to external services, like Azure, GitHub, Docker, Kubernetes, and many other services. Service Connections can be used to authenticate to these external services and to invoke diverse types of commands, like create and update resources in Azure, upload container images to Docker, or deploy applications to Kubernete ...
+
+To be able to invoke these commands, Service Connections need to have the right permissions to do so, for most types of Service Connections the permissions can be scoped to a subset of resources to limit the access they have. To improve the the principle of least privilege, it's often very common to have separate Service Connections for different environments like Dev/Test/QA/Prod.
```
```suggestion
To be able to invoke these commands, Service Connections need to have the right permissions to do so, for most types of Service Connections the permissions can be scoped to a subset of resources to limit the access they have. To improve the principle of least privilege, it's often very common to have separate Service Connections for different environments like Dev/Test/QA/Prod.
```

BlueBasher commented on docs/continuous-integration/dev-sec-ops/azure-devops/service-connection-security.md:
```diff
@@ -0,0 +1,49 @@
+
+To be able to invoke these commands, Service Connections need to have the right permissions to do so, for most types of Service Connections the permissions can be scoped to a subset of resources to limit the access they have. To improve the the principle of least privilege, it's often very common to have separate Service Connections for different environments like Dev/Test/QA/Prod.
+
+Because Service Connections have all the necessary permissions in these external services, it is crucial to secure Service Connections so they cannot be misused by accident or by malicious users.  
```
Thanks, good point. I've moved the section and added an example Pipeline of how the Service Connection could be misused.

shiranr reviewed (approved):
LGTM, thank you for contributing.

TessFerrandez commented on docs/continuous-integration/dev-sec-ops/azure-devops/service-connection-security.md:
```diff
@@ -0,0 +1,66 @@
+# Azure DevOps Service Connection Security
+
+In Azure DevOps, Service Connections are used in Azure DevOps Pipelines to connect to external services, like Azure, GitHub, Docker, Kubernetes, and many other services. Service Connections can be used to authenticate to these external services and to invoke diverse types of commands, like create and update resources in Azure, upload container images to Docker, or deploy applications to Kubernete ...
```
```suggestion
Service Connections are used in Azure DevOps Pipelines to connect to external services, like Azure, GitHub, Docker, Kubernetes, and many other services. Service Connections can be used to authenticate to these external services and to invoke diverse types of commands, like create and update resources in Azure, upload container images to Docker, or deploy applications to Kubernetes.
```
